modules/REST/V1/Languages.php

```diff
@@ -0,0 +1,809 @@
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of array{
+	 *     term_id: int
+	 * }|array{
+	 *     slug: non-empty-string
+	 * }
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function get_item( $request ) {
+		$language = $this->get_language( $request );
+
+		if ( is_wp_error( $language ) ) {
+			return $language;
+		}
+
+		return $this->prepare_item_for_response( $language, $request );
+	}
+
+	/**
+	 * Updates one language from the collection.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of array{
+	 *     term_id: int,
+	 *     slug?: non-empty-string,
+	 *     locale?: non-empty-string,
+	 *     name?: non-empty-string,
+	 *     is_rtl?: bool,
+	 *     term_group?: int,
+	 *     flag_code?: non-empty-string
+	 * }
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function update_item( $request ) {
+		$language = $this->get_language( $request );
+
+		if ( is_wp_error( $language ) ) {
+			return $language;
+		}
+
+		$prepared = (array) $this->prepare_item_for_database( $request );
+
+		if ( ! empty( $prepared ) ) {
+			if ( ! isset( $prepared['slug'], $prepared['locale'], $prepared['name'], $prepared['rtl'], $prepared['term_group'] ) ) {
+				$prepared = array_merge(
+					array(
+						'slug'       => $language->slug,
+						'locale'     => $language->locale,
+						'name'       => $language->name,
+						'rtl'        => (bool) $language->is_rtl,
+						'term_group' => $language->term_group,
+					),
+					$prepared
+				);
+			}
+			/**
+			 * @phpstan-var array{
+			 *     lang_id: int,
+			 *     slug: non-empty-string,
+			 *     locale: non-empty-string,
+			 *     name: non-empty-string,
+			 *     rtl: bool,
+			 *     term_group: int,
+			 *     flag?: non-empty-string
+			 * } $prepared
+			 */
+			$update = $this->languages->update( $prepared );
+
+			if ( is_wp_error( $update ) ) {
+				return $this->add_status_to_error( $update );
+			}
+
+			/** @var PLL_Language */
+			$language = $this->languages->get( $language->term_id );
+		}
+
+		return $this->prepare_item_for_response( $language, $request );
+	}
+
+	/**
+	 * Deletes one language from the collection.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of array{
+	 *     term_id: int
+	 * }
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function delete_item( $request ) {
+		$language = $this->get_language( $request );
+
+		if ( is_wp_error( $language ) ) {
+			return $language;
+		}
```

Comment:
Shouldn't we call `add_status_to_error()`?

Reply:
Same as above.